packages/mongo/oplog_observe_driver.js

```diff
@@ -486,6 +489,11 @@ _.extend(OplogObserveDriver.prototype, {
       // but fetch() yields.
       Meteor.defer(finishIfNeedToPollQuery(function () {
         while (!self._stopped && !self._needToFetch.empty()) {
+          // If we need to fetch more than a thousand documents, re-running is probably quicker
+          if (self._needToFetch.size() >= 1000) {
+            self._needToPollQuery();
+          }
+
```

Comment:
@mitar 

Here we decide to bail out on fetching 1000 documents via individual calls to MongoDB for each document and instead fetch all the needed documents in one operation.

On line 515 below you can see that the _docFetcher will be called for every document which 'currently needs to be fetched', which is the cause of the Fibers spike.

ps I'm not sure that this is already the perfect solution, but I am 100% sure that we have isolated the cause and should be working on ways to resolve it.

pps an alternative approach would be:
- batch Oplog messages (otherwise Meteor can never find out when to fetch multiple documents in one operation);
- pass an Array of _ids to the _docFetcher (basically rewriting the _docFetcher to fetch multiple docs).

But we will need to discuss this with people with more intricate knowledge of how this has been setup.